packages/core/src/sort-style-rules.ts

```diff
@@ -1,43 +1,83 @@
-import type { AtomicStyleResult, ConditionDetails } from '@pandacss/types'
+import type { AtRuleCondition, AtomicStyleResult, ConditionDetails, SelectorCondition } from '@pandacss/types'
 import { sortAtRules } from './sort-at-rules'
 
-const hasAtRule = (conditions: ConditionDetails[]) => conditions.some((details) => details.type === 'at-rule')
+const hasAtRule = (conditions: ConditionDetails[]) =>
+  conditions.some((details) => details.type === 'at-rule' || details.type === 'mixed')
 const styleOrder = [':link', ':visited', ':focus-within', ':focus', ':focus-visible', ':hover', ':active']
 const pseudoSelectorScore = (selector: string) => {
   const index = styleOrder.findIndex((pseudoClass) => selector.includes(pseudoClass))
   return index + 1
 }
-const compareConditions = (a: WithConditions, b: WithConditions) => {
-  if (a.conditions!.length === b.conditions!.length) {
-    const selector1 = a.conditions![0].value
-    const selector2 = b.conditions![0].value
+const compareSelectors = (a: WithConditions, b: WithConditions) => {
+  const aConds = a.conditions! as SelectorCondition[]
+  const bConds = b.conditions! as SelectorCondition[]
+
+  if (aConds.length === bConds.length) {
+    const selector1 = aConds[0].value
+    const selector2 = bConds[0].value
+
     return pseudoSelectorScore(selector1) - pseudoSelectorScore(selector2)
   }
 
-  return a.conditions!.length - b.conditions!.length
+  return aConds.length - bConds.length
 }
-const compareAtRuleConditions = (a: WithConditions, b: WithConditions) => {
-  if (a.conditions!.length === b.conditions!.length) {
-    const lastA = a.conditions![a.conditions!.length - 1]
-    const lastB = b.conditions![b.conditions!.length - 1]
 
-    const atRule1 = lastA.params ?? lastA.rawValue
-    const atRule2 = lastB.params ?? lastB.rawValue
+const flatten = (conds: ConditionDetails[]) => conds.flatMap((cond) => (cond.type === 'mixed' ? cond.value : cond))
 
-    if (!atRule1 || !atRule2) return 0
+const compareAtRuleOrMixed = (a: WithConditions, b: WithConditions) => {
```

Comment:
Can you include some tests that cover this new use case? It's a little hard to follow, and I prefer to read the tests instead.

If we could even isolate this bit of the code so we can test it independently, that would be even better.

Reply:
I had done that [here](https://github.com/chakra-ui/panda/blob/ae20e36145e4fddea5a9dd53627389ee93e48743/packages/core/__tests__/rule-processor.test.ts#L1296)

I just added a new one specifically for `compareAtRuleOrMixed` [here](https://github.com/chakra-ui/panda/blob/a0a3c1ee028d85b8ed9390f98cf3111a1b1d153d/packages/core/__tests__/conditions.test.ts#L93)

Reply:
added further improvements to the sorting logic & added a bunch more tests in [this commit](https://github.com/chakra-ui/panda/pull/2177/commits/7d821dbd5f2bc2ce03f8448471e25fc745954bd1) in the rule-processor.test file